clcache/__main__.py

```diff
@@ -116,6 +116,44 @@ def normalizeBaseDir(baseDir):
         return None
 
 
+class SuspendTracker():
```

Comment:
With this fix do I still need
<TrackFileAccess>false</TrackFileAccess>
in vs project files ?

Reply:
hm, I had that set to false in all my projects and incremental builds worked if I have not use clcache, that is why I am confused.. without disabling TrackFileAccess in all projects I had issues that when I do full rebuild VS2017 deleted while cache because all those files were created during previous build, I though that is why TrackFileAccess is needed. Ok, seems I will have to try.

By the way, your PR is not rebased on frerich master, do you have any branch that contains all his latest changes from master and your fixes ? that would save me some time..

Reply:
Ok, kind of progress here, you are right, with TrackFileAccess=false the incremental build does not work, very bad behavior. If I remove TrackFileAccess=false it almost work as I want, the problem is if you try to rebuild from visual studio, I noticed the objects survive clean, but json manifest get deleted, I think we need to protect them too.
Are you maybe using memcached, guess with memcached all problems related to this are gone, right ?

Reply:
> I noticed the objects survive clean, but json manifest get deleted,  I think we need to protect them too.
They are protected, probably not in all cases, could you please attach verbose log of clean target. 
> Are you maybe using memcached, guess with memcached all problems related to this are gone, right ? 
No, it should work without memcached.

Reply:
So, here I am. The clcache I am using is made with pip install git+https://github.com/9a4gl/clcache.git .. it is original one with your changes on top, you can check that. By the way, I am not python guy, I was trying to compile with pyinstaller clcache.py, but it complains there is no clcache.py, what is correct way to compile and install from command line in cloned directory ? That way I can debug with traces at least, now I am stuck with exe I am really not sure it is the one I created.

Attached are logs and it is clear jsons are deleted. Hope this one will turn some light on your side.

Reply:
python 3.7.1 64-bit, hm, I was suspecting that maybe FileTracker64.dll is used that is not covered with your patch. I need this verision of python because other stuff, will try to use 64bit toolchain, maybe that will help. clcache-msbuild-install I used, just from his other repo, didn't know he actually moved his work to new repo, will try that.

Still, what is proper way to compile clcache, pyinstaller clcache.py just do not work ?

Reply:
> Still, what is proper way to compile clcache, pyinstaller clcache.py just do not work ?

`pyinstaller pyinstaller\clcache_main.py`

Reply:
Aha, I have solved the issue, 5 minutes of my python glory. using x86 toolchains and x64 python. So, here is my fix, you can append it to your fix:
https://github.com/9a4gl/clcache/commit/dd15bd8f6faeebd8a4d3a0a677737bd07812fe5b

Reply:
Done https://github.com/9a4gl/clcache/commit/5597e1dca2824ff4fc07f19929cabecf861278b6

Ah, msbuild support is needed. What is the reason for that ?

We need to integrate clcache in small company and we would like to have common cache, does it work well with network share ? Guess not. To me it looks like something like dropbox would do the job. Do you know any free/open software that do replica of cache and not cause conflicts becasue of updating stat.txt and other stuff so often ?

Reply:
Probably memcached backend can be used for shared cache.